src/network/lan.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+Copyright (C) 2016 proller <proler@gmail.com>
+*/
+
+/*
+This file is part of Freeminer.
+
+Freeminer is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Freeminer  is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Freeminer.  If not, see <http://www.gnu.org/licenses/>.
```

Comment:
Could the licensing situation around the file be clarified in the actual source code itself somehow? It is my understanding the author of the Freeminer has given permission for the LAN discovery to be included in Minetest (presumably under the terms of the LGPL license?) but that wouldn't be clear to anyone obtaining a copy of the Minetest source code down the road.

Reply:
Yes, this needs to be updated now that it is in minetest. @proller, how should this be fixed?

Reply:
replace this with minetest header

Reply:
How's this:
/*
Minetest
Copyright (C) 2024 proller proller's email

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation; either version 2.1 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License along
with this program; if not, write to the Free Software Foundation, Inc.,
51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
*/

Would it be right to put my own name after @proller's name in the Copyright section?